types/src/main.ts

```diff
@@ -4087,6 +4087,125 @@ export namespace InlayHint {
 	}
 }
 
+/**
+ * An inline completion item represents a text snippet that is proposed inline to complete text that is being typed.
+ *
+ * @since 3.18.0
+ */
+export interface InlineCompletionItem {
+	/**
+	 * The text to replace the range with. Must be set.
+	 */
+	insertText: string;
+
+	/**
+	 * A text that is used to decide if this inline completion should be shown. When `falsy` the {@link InlineCompletionItem.insertText} is used.
+	 */
+	filterText?: string;
+
+	/**
+	 * The range to replace. Must begin and end on the same line.
+	 */
+	range?: Range;
+
+	/**
+	 * An optional {@link Command} that is executed *after* inserting this completion.
+	 */
+	command?: Command;
+
+	/**
+	 * The format of the insert text. The format applies to the `insertText`. If omitted defaults to `InsertTextFormat.PlainText`.
+	 */
+	insertTextFormat?: InsertTextFormat;
```

Comment:
Moved. Happy to alter this to use the complex type instead.

Reply:
@aeschli friendly ping. I am actually leaning more towards adding the `StringContent` type.

Reply:
Yes, agree, makes sense to avoid the additional property as don't need to be backward compatible.
In `vscode.d.ts` there's the `SnippetString`.
So I suggest: `insertText: string | SnippetString;`

Reply:
Done.

Reply:
Added the SnippetString type, removed the InsertTextFormat usage, and updated the Inline Completions to use the new type.